Bottom: e9731220dc88a3d073f7178c9a08e98986b6cadd
Top:    7e378d5d514ce455f6abbb773533f8f585a9b0e8
Author: Ian Kent <raven@themaw.net>
Date:   2013-07-30 16:11:57 +0800

autofs-5.0.7 - fix dumpmaps multi output

If the maps contain the old style "multi" type the dumped map information
isn't right.


---

diff --git a/lib/master.c b/lib/master.c
index eca3523..49d4691 100644
--- a/lib/master.c
+++ b/lib/master.c
@@ -1281,6 +1281,25 @@ static void list_source_instances(struct map_source *source, struct map_source *
 	return;
 }
 
+static void print_map_info(struct map_source *source)
+{
+	int i = 0;
+
+	if (source->argv[0] && *source->argv[0] != '-') {
+		printf("  map: %s\n", source->argv[0]);
+		i = 1;
+	}
+
+	if (source->argc > 1) {
+		printf("  arguments: ");
+		for (; i < source->argc; i++)
+			printf("%s ", source->argv[i]);
+		printf("\n");
+	}
+
+	return;
+}
+
 int master_show_mounts(struct master *master)
 {
 	struct list_head *p, *head;
@@ -1357,20 +1376,10 @@ int master_show_mounts(struct master *master)
 			}
 
 			if (source->argc >= 1) {
-				i = 0;
-				if (source->argv[0] && *source->argv[0] != '-') {
-					printf("  map: %s\n", source->argv[0]);
-					i = 1;
-				}
+				output_map_info(source);
 				if (count && ap->type == LKP_INDIRECT)
 					printf("  duplicate indirect map entry"
 					       " will be ignored at run time\n");
-				if (source->argc > 1) {
-					printf("  arguments: ");
-					for (; i < source->argc; i++)
-						printf("%s ", source->argv[i]);
-					printf("\n");
-				}
 			}
 
 			printf("\n");
